### PR TITLE
Update API to V4

### PR DESCRIPTION
**Pull request by dfernandezramos**
Since some models affect different test classes I've done a single PR separating each endpoint group/endpoint tests class by commits.

### CODE REVIEW

**Comment by dfernandezramos** [comment]
- Failing tests in master are still failing but I've checked that my changes don't affect them.
- GetChampionMasteryAsyncTest () on ChampionMasteryTests has a commented Assert line since Riot is returning wrong data (check difference between v3 and v4). The behavior should be the same but the id encryption is wrong or something like that.
- I've changed some tests requests since old requests were asking for old data. I've updated them.
- Since summoners id's and accountId's are encrypted now depending on the API key which has made the request, I've made the tests that need Id's make a previous request by summoner name to get the encrypted data.

File: RiotNet.Tests/Apis/ChampionMasteryTests.cs
Comment (aj-r) [inline RiotNet.Tests/Apis/ChampionMasteryTests.cs:37]: Add a comment explaining why this line is commented out

File: RiotNet.Tests/Apis/TournamentTests.cs
Comment (aj-r) [inline RiotNet.Tests/Apis/TournamentTests.cs]: I would prefer that variable names don't contain underscores (`rndmInternetManId` is fine)

File: RiotNet/Models/Summoner.cs
Comment (aj-r) [inline RiotNet/Models/Summoner.cs]: Only capitalize the first letter of acronyms (`Puuid`)

**Review by aj-r — CHANGES_REQUESTED** [review]
Looks great! Just a few minor things.

File: RiotNet.Tests/Apis/ChampionMasteryTests.cs
Comment (dfernandezramos) [reply RiotNet.Tests/Apis/ChampionMasteryTests.cs:37]: done

File: RiotNet.Tests/Apis/TournamentTests.cs
Comment (dfernandezramos) [reply RiotNet.Tests/Apis/TournamentTests.cs]: done

Comment (dfernandezramos) [reply RiotNet.Tests/Apis/TournamentTests.cs]: done

File: RiotNet/Models/Summoner.cs
Comment (dfernandezramos) [reply RiotNet/Models/Summoner.cs]: done

**Comment by dfernandezramos** [comment]
Changes processed. I'll merge the fixup commits with each proper commit after approve 👍

File: RiotNet.Tests/Apis/ChampionMasteryTests.cs
Comment (aj-r) [inline RiotNet.Tests/Apis/ChampionMasteryTests.cs]: Typo: known

**Review by aj-r — CHANGES_REQUESTED** [review]

File: RiotNet.Tests/Apis/ChampionMasteryTests.cs
Comment (aj-r) [reply RiotNet.Tests/Apis/ChampionMasteryTests.cs]: Also: encrypted

Comment (dfernandezramos) [reply RiotNet.Tests/Apis/ChampionMasteryTests.cs]: Done

**Comment by dfernandezramos** [comment]
All changes processed. Since your last changes request were two typos, I've already merged the fixup commits with their respective ones. 👍

**Review by aj-r — APPROVED** [review]

**Comment by dfernandezramos** [comment]
Please let me know if you need something else. I assume you will merge this PR when you can.

**Comment by aj-r** [comment]
Everything is fine, I just wanted to double-check that you were merging to the right branch first (which I can't do on my phone for some reason). Looks good!

**Pull request merged**
